app/models/favorites.php

```diff
@@ -0,0 +1,63 @@
+<?php
+class Favorites extends AppModel
+{
+    public static function vote($user_id, $comment_id, $action)
+    {
+        $db = DB::conn();
+        try {
+            $db->begin();
+            if ($action === 'favorite') {
+                self::insertFavorites($user_id, $comment_id);
+            } else {
+                self::deleteFavorites($user_id, $comment_id);
```

Comment:
Since in your functions "insertFavorites" & "deleteFavorites" you already connect to DB, why still connect to DB in this vote function? :sweat_smile: I think you have to re-check each lines of codes you add to ensure code quality :bow: